src/contacts.cs

```diff
@@ -449,6 +1237,213 @@ interface CNLabelContactRelationKey {
+	}
+
+	[Watch (6,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (CNChangeHistoryEvent))]
+	[DisableDefaultCtor]
+	interface CNChangeHistoryAddGroupEvent
+	{
+		[Export ("group", ArgumentSemantic.Strong)]
+		CNGroup Group { get; }
+
+		[Export ("containerIdentifier", ArgumentSemantic.Strong)]
+		string ContainerIdentifier { get; }
+	}
+
+	[Watch (6,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (CNChangeHistoryEvent))]
+	[DisableDefaultCtor]
+	interface CNChangeHistoryUpdateGroupEvent
+	{
+		[Export ("group", ArgumentSemantic.Strong)]
+		CNGroup Group { get; }
+	}
+
+	[Watch (6,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (CNChangeHistoryEvent))]
+	[DisableDefaultCtor]
+	interface CNChangeHistoryDeleteGroupEvent
+	{
+		[Export ("groupIdentifier", ArgumentSemantic.Strong)]
+		string GroupIdentifier { get; }
+	}
+
+	[Watch (6,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (CNChangeHistoryEvent))]
+	[DisableDefaultCtor]
+	interface CNChangeHistoryAddMemberToGroupEvent
+	{
+		[Export ("member", ArgumentSemantic.Strong)]
+		CNContact Member { get; }
+
+		[Export ("group", ArgumentSemantic.Strong)]
+		CNGroup Group { get; }
+	}
+
+	[Watch (6,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (CNChangeHistoryEvent))]
+	[DisableDefaultCtor]
+	interface CNChangeHistoryRemoveMemberFromGroupEvent
+	{
+		[Export ("member", ArgumentSemantic.Strong)]
+		CNContact Member { get; }
+
+		[Export ("group", ArgumentSemantic.Strong)]
+		CNGroup Group { get; }
+	}
+
+	[Watch (6,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (CNChangeHistoryEvent))]
+	[DisableDefaultCtor]
+	interface CNChangeHistoryAddSubgroupToGroupEvent
+	{
+		[Export ("subgroup", ArgumentSemantic.Strong)]
+		CNGroup Subgroup { get; }
+
+		[Export ("group", ArgumentSemantic.Strong)]
+		CNGroup Group { get; }
+	}
+
+	[Watch (6,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (CNChangeHistoryEvent))]
+	[DisableDefaultCtor]
+	interface CNChangeHistoryRemoveSubgroupFromGroupEvent
+	{
+		[Export ("subgroup", ArgumentSemantic.Strong)]
+		CNGroup Subgroup { get; }
+
+		[Export ("group", ArgumentSemantic.Strong)]
+		CNGroup Group { get; }
+	}
+
+	[Watch (6,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CNFetchRequest {}
+
+	[Watch (6,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (CNFetchRequest))]
+	interface CNChangeHistoryFetchRequest : NSSecureCoding
+	{
+		[NullAllowed, Export ("startingToken", ArgumentSemantic.Copy)]
+		NSData StartingToken { get; set; }
+
+		[NullAllowed, Export ("additionalContactKeyDescriptors", ArgumentSemantic.Copy)]
+		// we cannot use ICNKeyDescriptor as Apple (and others) can adopt it from categories
+		// cannot be exposed as NSString since they could be internalized types, like CNAggregateKeyDescriptor
+		NSArray AdditionalContactKeyDescriptors { get; set; }
```

Comment:
What about `NSObject []`? at least more C# friendly

Reply:
I don't quite understand why you can't use `ICNKeyDescriptor[]` - that's how it's defined in the header, so it should work.

Reply:
Updated to `ICNKeyDescriptor[]` here.